lms/djangoapps/courseware/access.py

```diff
@@ -371,9 +397,12 @@ def _can_load_course_overview(user, course_overview):
         load access.
     """
     return (
-        not course_overview.visible_to_staff_only
-        and _can_access_descriptor_with_start_date(user, course_overview, course_overview.id)
-    ) or _has_staff_access_to_descriptor(user, course_overview, course_overview.id)
+        _has_staff_access_to_descriptor(user, course_overview, course_overview.id) or
+        (
+            _visible_to_nonstaff_users(course_overview)
+            and _can_access_descriptor_with_start_date(user, course_overview, course_overview.id)
+        )
+    )
```

Comment:
Please rearrange the logic here so the staff access check happens at the end (as it was before).  I believe the staff access check currently requires a sql database hit.